tests/Controller/Api/External/ConfigControllerTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the mangel.io project.
+ *
+ * (c) Florian Moser <git@famoser.ch>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Tests\Controller\Api\External;
+
+use App\Service\Interfaces\PathServiceInterface;
+use App\Tests\Controller\Api\External\Base\ApiController;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class ConfigControllerTest extends ApiController
+{
+    /**
+     * @var string
+     */
+    private $configFilePath;
+
+    /**
+     * ConfigControllerTest constructor.
+     *
+     * @param string|null $name
+     * @param array $data
+     * @param string $dataName
+     */
+    public function __construct(?string $name = null, array $data = [], string $dataName = '')
+    {
+        parent::__construct($name, $data, $dataName);
+
+        self::bootKernel();
+        $pathService = self::$container->get(PathServiceInterface::class);
+
+        $this->configFilePath = $pathService->getTransientFolderRoot() . \DIRECTORY_SEPARATOR . 'domainOverrides.json';
+    }
+
+    /**
+     * tests the login functionality.
+     */
+    public function testDomainOverrides_configExists_returnsConfig()
+    {
+        $client = static::createClient();
+        $this->createConfigFile();
+
+        $response = $this->doConfigRequest($client);
+        $trialResponse = json_decode($response->getContent());
+        $this->assertNotNull($trialResponse->domainOverrides);
+
+        $domainOverrides = $trialResponse->domainOverrides;
+        $this->assertTrue(\is_array($domainOverrides));
+        $this->assertNotEmpty($domainOverrides);
+
+        $entry = $domainOverrides[0];
+        $this->assertNotNull($entry->userInputDomain);
+        $this->assertNotNull($entry->serverUrl);
+        $this->assertNotNull($entry->userLoginDomain);
+
+        $this->removeConfigFile();
+    }
+
+    /**
+     * tests the login functionality.
+     */
+    public function testDomainOverrides_configDoesNotExist_returnsNotFound()
+    {
+        $client = static::createClient();
+        $this->removeConfigFile();
+
+        $this->expectException(NotFoundHttpException::class);
+        $this->doConfigRequest($client);
+    }
+
+    private function createConfigFile()
+    {
+        file_put_contents($this->configFilePath, '{
+  "domainOverrides": [
+    {
+      "userInputDomain":"mangel.io",
+      "serverUrl":"https://app.mangel.io",
```

Comment:
We've established the standard of casing all-caps acronyms homogeneously, i.e. this should be `serverURL`.